app/models/katello/sync_plan.rb

```diff
@@ -40,25 +41,28 @@ def product_enabled
     end
 
     def save_with_logic!
+      ForemanTasks::RecurringLogic.valid_cron_logic? self.cron_expression
```

Comment:
So many failures.. :open_mouth: Forgot to push this change in foreman_tasks..
Question: Is it ok to add a class method to recurring_logic class to check if the recurring logic is correct?
```
def self.valid_cron_logic?(cron_logic)
   parser = CronParser.new(cron_logic, Time.zone)
   parser.next(Time.zone.now)
   rescue ArgumentError => _
      false
end

```

Reply:
I think thats okay to put in the sync plan class if thats what you mean.